### DISCUSSION TITLE
How to Detach an XDP Program and Unload It from the Kernel Using the `ebpf` Library？

### DISCUSSION THREAD

**Discussion by binw666** [cilium/ebpf#1565] [Q&A]
I have a network interface on my host, and I obtained the ID of the XDP program on the interface via `netlink`. I want to detach this program and unload both the program and the map it uses from the kernel. How should I proceed?

**Comment by kwakubiney** [comment]
To unload an eBPF program, you have to make sure there are no more references to the loaded program. One step towards your goal is to detach it from your network link with `netlink.LinkSetXdpFd(link, -1)`
